### ISSUE TITLE
Deleting shader!

### ISSUE THREAD

**Issue by ghost** [TheCherno/Sparky#39]
When you load the vertex and fragment shader you try to [delete](https://github.com/TheCherno/Sparky/blob/master/Sparky-core/src/graphics/shader.cpp#L65) them but they never(Until you close the window) are going to be deleted because they need to be detached first from the program.

Source: [Stackoverflow](http://gamedev.stackexchange.com/questions/47910/after-a-succesful-gllinkprogram-should-i-delete-detach-my-shaders)

**Comment by Shimmen** [comment]
This is true, but should be an easy fix. Before [this](https://github.com/TheCherno/Sparky/blob/master/Sparky-core/src/sp/graphics/shaders/Shader.cpp#L167) line, simply append:

``` c++
glDetachShader(program, vertex);
glDetachShader(program, fragment);
```

This must be done since deleting a shader doesn't automatically detach it from the program (as described [here](http://docs.gl/gl3/glDetachShader)).

I could create a pull request if it would be appreciated.
